src/main/java/com/management/HumanResources/service/TimeOffService.java

```diff
@@ -1,68 +1,149 @@
 package com.management.HumanResources.service;
 
-import java.time.*;
+import java.time.DayOfWeek;
+import java.util.*;
 
+import com.management.HumanResources.exceptions.*;
 import com.management.HumanResources.model.*;
 
+import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.stereotype.Service;
 
 @Service
 public class TimeOffService {
-    
-    private final double DAILY_WORK_HOURS = 8; // Max time off hours that can be used per day.
-    private final double WEEKLY_WORK_DAYS = 5; // Max time off days that can be used per week.
 
-    //TODO: This class will hold the logic that checks for time off conflicts in the schedule
+    @Autowired ScheduleService scheduleService;
+    
+    private final int MIN_WORKERS = 3; // Minimum number of employees that work at a given time.
 
-    public void approveTimeOffForEmployee(TimeOff timeOff, EmployeeTime employeeTime)
-    {
+    public void approveTimeOffForEmployee(TimeOff timeOff, EmployeeTime employeeTime) {
         timeOff.setApproved(true);
         timeOff.setReviewed(true);
-        employeeTime.setHoursRemaining(employeeTime.getHoursRemaining() - getRealNumberOfTimeOffHours(timeOff));
+        employeeTime.setHoursRemaining(employeeTime.getHoursRemaining() - getNumberOfTimeOffHoursForEmployee(timeOff, employeeTime));
     }
 
-    public void declineTimeOff(TimeOff timeOff)
-    {
+    public void declineTimeOff(TimeOff timeOff) {
         timeOff.setApproved(false);
         timeOff.setReviewed(true);
     }
 
     /**
-     * Checks if the employee has enough remaining time off hours for the requested time off.
+     * Checks if a time off is valid for an employee and throws an exception if
+     * invalid.
+     * 
+     * @throws InvalidTimeOffException
      */
-    public boolean isValidTimeOffForEmployee(TimeOff timeOff, EmployeeTime employeeTime) {
-        return employeeTime.getHoursRemaining() >= getRealNumberOfTimeOffHours(timeOff);
+    public void validateTimeOffForEmployee(TimeOff timeOff, EmployeeTime employeeTime) throws InvalidTimeOffException {
+        
+        // Check if the time off start and end are in the correct order.
+        if (timeOff.isStartBeforeEnd()) {
```

Comment:
The naming of this method doesn't match what it's doing here. When I read this it says to me "If the start time is before the end time, throw an exception"
But for time off the start is supposed to be before the end.

Reply:
Nice catch! I just forgot to put the negation operator.